src/utils/mfa/index.ts

```diff
@@ -127,11 +134,44 @@ async function postRequest(
   );
 }
 
-function errorHandler(error: any, trace: string): void {
+async function putRequest(
+  updateInput: UpdateInformationInput,
+  sessionDetails: UpdateInformationSessionValues,
+  http: Http = new Http(getMfaServiceUrl())
+): Promise<{
+  status: number;
+  data: MfaMethod;
+}> {
+  const response = await http.client.put<MfaMethod>(
+    format(
+      METHOD_MANAGEMENT_API.MFA_METHODS_PUT,
+      updateInput.mfaMethod.mfaIdentifier
+    ),
+    {
+      email: updateInput.email,
+      credential: updateInput.updatedValue,
+      otp: updateInput.otp,
+      mfaMethod: updateInput.mfaMethod,
+    },
+    getRequestConfig({
+      token: sessionDetails.accessToken,
+      validationStatuses: [
+        HTTP_STATUS_CODES.OK,
+        HTTP_STATUS_CODES.NOT_FOUND,
+        HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR,
+      ],
+      ...sessionDetails,
+    })
+  );
+
+  return response;
+}
+
+function errorHandler(error: any, trace: string, action: string): void {
   if (!error.response || !error.response.status) {
     logger.error(
       { trace },
-      `Failed to retrieve from MFA endpoint ${error.message}`
+      `Failed to default ${action} MFA endpoint ${JSON.stringify(error)}`
```

Comment:
This is useful for us but we should try avoiding to stringify the whole error as it could contain session information and user data.

Reply:
yup, got it. Definitely left that there by mistake. used when debugging. Now Addressed.